src/freenet/node/RequestSender.java

```diff
@@ -939,10 +939,8 @@ protected MessageFilter makeAcceptedRejectedFilter(PeerNode next,
 		MessageFilter mfRejectedLoop = MessageFilter.create().setSource(next).setField(DMT.UID, uid).setTimeout(acceptedTimeout).setType(DMT.FNPRejectedLoop);
 		MessageFilter mfRejectedOverload = MessageFilter.create().setSource(next).setField(DMT.UID, uid).setTimeout(acceptedTimeout).setType(DMT.FNPRejectedOverload);
 		
-		// mfRejectedOverload must be the last thing in the or
-		// So its or pointer remains null
-		// Otherwise we need to recreate it below
-		return mfAccepted.or(mfRejectedLoop.or(mfRejectedOverload));
+		// the order of these filters is performance critical. The last or-filter is checked first.
```

Comment:
~~This seems really weird. Looking at the filter creation above it’s absolutely unclear _why_ these filters should show any difference in performance, they are all created the same way, only with a different message type. Now, I haven’t timed any of this but if the reordering of the filters changes the overall performance of this method, it’s probably _not_ because the filters themselves have differing performances but because the most-matched filter was checked last and thus many useless matchers were made, in which casethe comment should reflect that.~~

Okay, re-read the comment, it’s not what it said. 🙂 Anyway, if measuring has shown that reordering the filters increased performance because many useless matches were not made, the comment should say _that_, if anything. 😄

Reply:
I added a second line now: **So the last filter in the or-"chain" must be the filter which matches most frequently**